internal/services/netapp/netapp_account_encryption_resource.go

```diff
@@ -0,0 +1,342 @@
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.NetApp.AccountClient
+			keyVaultsClient := metadata.Client.KeyVault
+			resourcesClient := metadata.Client.Resource
+
+			id, err := netappaccounts.ParseNetAppAccountID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("Decoding state for %s", id)
+			var state netAppModels.NetAppAccountEncryption
+			if err := metadata.Decode(&state); err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("Updating %s", id)
+
+			update := netappaccounts.NetAppAccountPatch{
+				Properties: &netappaccounts.AccountProperties{},
+			}
+
+			if metadata.ResourceData.HasChange("user_assigned_identity_id") || metadata.ResourceData.HasChange("system_assigned_identity_principal_id") || metadata.ResourceData.HasChange("encryption") {
+				encryptionExpanded, err := expandEncryption(ctx, state.Encryption, keyVaultsClient, resourcesClient, pointer.To(state))
+				if err != nil {
+					return err
+				}
+
+				update.Properties.Encryption = encryptionExpanded
+
+				if err := client.AccountsUpdateThenPoll(ctx, pointer.From(id), update); err != nil {
+					return fmt.Errorf("updating %s: %+v", id, err)
+				}
+
+				metadata.SetID(id)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r NetAppAccountEncryptionResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+
+			client := metadata.Client.NetApp.AccountClient
+
+			id, err := netappaccounts.ParseNetAppAccountID((metadata.ResourceData.Id()))
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("Decoding state for %s", id)
+			var state netAppModels.NetAppAccountEncryption
+			if err := metadata.Decode(&state); err != nil {
+				return err
+			}
+
+			existing, err := client.AccountsGet(ctx, pointer.From(id))
+			if err != nil {
+				if existing.HttpResponse.StatusCode == http.StatusNotFound {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %v", id, err)
+			}
+
+			if existing.Model.Properties.Encryption == nil {
+				return fmt.Errorf("encryption information does not exist for %s", id)
+			}
+
+			anfAccountIdentityFlattened, err := identity.FlattenLegacySystemAndUserAssignedMapToModel(existing.Model.Identity)
+			if err != nil {
+				return err
+			}
+
+			model := netAppModels.NetAppAccountEncryption{
+				NetAppAccountID: id.ID(),
+				Encryption:      flattenEncryption(existing.Model.Properties.Encryption),
+			}
+
+			if len(anfAccountIdentityFlattened) > 0 {
+
+				if anfAccountIdentityFlattened[0].Type == identity.TypeSystemAssigned {
+					model.SystemAssignedIdentityPrincipalID = anfAccountIdentityFlattened[0].PrincipalId
+				}
+
+				if anfAccountIdentityFlattened[0].Type == identity.TypeUserAssigned {
+					model.UserAssignedIdentityID = anfAccountIdentityFlattened[0].IdentityIds[0]
```

Comment:
We should confirm the length `IdentityIds[0]` before accessing it to prevent a crash